README.md

```diff
@@ -2,152 +2,316 @@ Plop
+### 3. Create a plopfile.js at the root of your project
 ``` javascript
 module.exports = function (plop) {
 	// create your generators here
 	plop.setGenerator('basics', {
 		description: 'this is a skeleton plopfile',
-		prompts: [],
-		actions: []
+		prompts: [], // array of inquirer prompts
+		actions: []  // array of actions
 	});
 };
 ```
----
-# Anatomy of a Plop Generator
-The main parts of a plop generator are the plop file (`plopfile.js`) and the templates. Templates can either be inline, or in separate files.
 
-A basic plop file starts its life as a lowly node module that exports a function that accepts the `plop` object.
+## Your First Plopfile
+A plopfile starts its life as a lowly node module that exports a function that accepts the `plop` object as its first parameter.
+
 ``` javascript
 module.exports = function (plop) {};
 ```
-The `plop` object offers four main functions (`addHelper`, `addPartial`, `addPrompt`, `setGenerator`).
 
-## plop.addHelper(name, helper)
-- name {String}
-- helper {Function}
+The `plop` object exposes the plop api object which contains the `setGenerator(name, config)` function. This is the function that you use to (wait for it) create a generator for this plopfile. When `plop` is run from the terminal in this directory (or any sub-directory), a list of these generators will be displayed.
+
+Let's try setting up a basic generator to see how that looks.
+
+``` javascript
+module.exports = function (plop) {
+	// controller generator
+	plop.setGenerator('controller', {
+		description: 'application controller logic',
+		prompts: [{
+			type: 'input',
+			name: 'name',
+			message: 'controller name please'
+		}],
+		actions: [{
+			type: 'add',
+			path: 'src/{{name}}.js',
+			templateFile: 'plop-templates/controller.hbs'
+		}]
+	});
+};
+```
+
+The *controlller* generator we created above will ask us 1 question, and create 1 file. This can be expanded to ask as many questions as needed, and create as many files as needed. There are also additional actions that can be used to alter our codebase in different ways.
+
+## CLI Usage
+Once plop is installed, and you have created a generator, you are ready to run plop from the terminal. Running `plop` with no parameters will present you with a list of generators to pick from. You can also run `plop [generatorName]` to trigger a generator directly. If you did not install plop globally, you will need to setup an npm script to run plop for you.
+
+### Bypassing Prompts
+Once you get to know a project (and its generators) well, you may want to provide  answers to the prompts when you run the generator. If I have (for instance) a `component` generator that has one prompt (name), I can run that generator using `plop component "some component name"` and it will immediately execute as though I had typed "some component name" into the prompt. If that same generator had a second prompt, the same input would have resulted in the user being prompted for the second value.
```

Comment:
It looks like there are two spaces in between _provide  answers_.

I also feel that a screencast / gif will be perfect to illustrate what is explained here. What do you think?

Reply:
I completely agree. I'd like to add more media in general soon. I've been meaning to make a small video series introducing the different concepts and educating people on how to use the tool.

![soon](http://i0.kym-cdn.com/photos/images/facebook/000/117/021/enhanced-buzz-28895-1301694293-0.jpg)